test/unit_tests/augmentation/test_base.py

```diff
@@ -0,0 +1,156 @@
+from braindecode.datautil import create_from_mne_epochs
+
+
+def dummy_k_operation(X, y, k, *args, **kwargs):
+    return torch.ones_like(X) * k, y
+
+
+def common_tranform_assertions(input_batch, output_batch, expected_X=None):
+    """ Assert whether shapes and devices are conserved. Also, (optional)
+    checks whether the expected features matrix is produced.
+
+    Parameters
+    ----------
+    input_batch : tuple
+        The batch given to the transform containing a tensor X, of shape
+        (batch_sizze, n_channels, sequence_len), and a tensor  y of shape
+        (batch_size).
+    output_batch : tuple
+        The batch output by the transform. Should have two elements: the
+        transformed X and y.
+    expected_X : tensor, optional
+        The expected first element of output_batch, which will be compared to
+        it. By default None.
+    """
+    X, y = input_batch
+    tr_X, tr_y = output_batch
+    assert tr_X.shape == X.shape
+    assert tr_X.shape[0] == tr_y.shape[0]
+    assert torch.equal(tr_y, y)
+    assert X.device == tr_X.device
+    if expected_X is not None:
+        assert torch.equal(tr_X, expected_X)
+
+
+@pytest.mark.parametrize("k1,k2,expected,p1,p2", [
+    (1, 0, 0, 1, 1),  # replace by 1s with p=1, then 0s with p=1 -> 0s
+    (0, 1, 1, 1, 1),  # replace by 0s with p=1, then 1s with p=1 -> 1s
+    (1, 0, 1, 1, 0),  # replace by 1s with p=1, then 1s with p=0 -> 1s
+    (0, 1, 0, 1, 0),  # replace by 0s with p=1, then 0s with p=0 -> 0s
+    (1, 0, 0, 0, 1),  # replace by 1s with p=0, then 0s with p=1 -> 0s
+    (0, 1, 1, 0, 1),  # replace by 0s with p=0, then 1s with p=1 -> 1s
+])
+def test_transform_composition(random_batch, k1, k2, expected, p1, p2):
+    X, y = random_batch
+    dummy_transform1 = Transform(partial(dummy_k_operation, k=k1), p1)
+    dummy_transform2 = Transform(partial(dummy_k_operation, k=k2), p2)
+    concat_transform = Compose([dummy_transform1, dummy_transform2])
+    expected_tensor = torch.ones(
+        X.shape,
+        device=X.device
+    ) * expected
+
+    common_tranform_assertions(
+        random_batch,
+        concat_transform(X, y),
+        expected_tensor
+    )
+
+
+@pytest.mark.parametrize("probability", [0, 1])
+def test_transform_with_kwargs(random_batch, rng_seed, probability):
+    rng = check_random_state(rng_seed)
+    k = rng.randint(10)
+    dummy_transform = Transform(
+        dummy_k_operation,
+        probability=probability,
+        k=k
+    )
+    X, y = random_batch
+    expected_tensor = torch.ones(
+        X.shape,
+        device=X.device
+    ) * k if probability else X
+    common_tranform_assertions(
+        random_batch,
+        dummy_transform(X, y),
+        expected_tensor
+    )
+
+
+def test_transform_proba_exception(random_batch, rng_seed):
+    rng = check_random_state(rng_seed)
+    with pytest.raises(AssertionError):
+        k = rng.randint(10)
+        _ = Transform(
```

Comment:
Not really needed here no? Or maybe I am missing something.
```suggestion
        Transform(
```

Reply:
The idea was not to pollute the stdout during tests, but I guess that's not really a problem :)